mapper/pojo-standalone/src/main/java/org/hibernate/search/mapper/pojo/standalone/mapping/impl/StandalonePojoMapping.java

```diff
@@ -62,9 +61,12 @@ public class StandalonePojoMapping extends AbstractPojoMappingImplementor<Standa
 
 	private final SchemaManagementListener schemaManagementListener;
 
-	StandalonePojoMapping(PojoMappingDelegate mappingDelegate, StandalonePojoTypeContextContainer typeContextContainer,
+	StandalonePojoMapping(ConfigurationPropertySource configurationPropertySource, BeanResolver beanResolver,
+			PojoMappingDelegate mappingDelegate, StandalonePojoTypeContextContainer typeContextContainer,
 			SchemaManagementListener schemaManagementListener) {
 		super( mappingDelegate );
+		this.configurationPropertySource = configurationPropertySource;
```

Comment:
This is not a good idea, as this means the configuration (and all the bean instances it may contain) will never get garbage collected.

Can you try extracting the default strategy (as a `ConfiguredIndexingPlanSynchronizationStrategy`) in `#start`?